backend/danswer/search/pipeline.py

```diff
@@ -240,32 +245,33 @@ def _get_sections(self) -> list[InferenceSection]:
         merged_ranges = [
             merge_chunk_intervals(ranges) for ranges in doc_chunk_ranges_map.values()
         ]
+
         flat_ranges = [r for ranges in merged_ranges for r in ranges]
+        flattened_inference_chunks = []
+        parallel_functions_with_args = []
 
         for chunk_range in flat_ranges:
-            functions_with_args.append(
-                (
-                    # If Large Chunks are introduced, additional filters need to be added here
-                    self.document_index.id_based_retrieval,
+            if chunk_range.start == chunk_range.end:
```

Comment:
can we add a comment here explaining why we can do this (e.g. no need to fetch anything additional from vespa if there is no range, so this improves latency)

Reply:
actually, I don't think we need to ever do any retrieval as long as `above` / `below` are both 0